ADAL/src/cache/ADTokenCache.m

```diff
@@ -567,7 +603,7 @@ - (BOOL)addOrUpdateImpl:(ADTokenCacheItem *)item
 
 - (NSString *)description
 {
-    return [NSString stringWithFormat:@"ADTokenCache: %@", _cache];
+    return [NSString stringWithFormat:@"ADTokenCache: %@", [self cacheCopy]];
```

Comment:
I could certainly add the `@synchronize` if you want. I tend to prefer readability, especially in a method like `description` which is rarely called and should never be a performance problem. It's up to you, however.